controllers/virtualmachinesnapshot/virtualmachinesnapshot_controller.go

```diff
@@ -190,49 +191,26 @@ func (r *Reconciler) ReconcileNormal(ctx *pkgctx.VirtualMachineSnapshotContext)
 		return ctrl.Result{}, errors.New("VM hasn't been created and has no uniqueID")
 	}
 
-	// Calculate the requested capacity of the snapshot at the beginning only once.
 	if vmSnapshot.Status.Storage == nil {
 		vmSnapshot.Status.Storage = &vmopv1.VirtualMachineSnapshotStorageStatus{}
 	}
 
+	// Calculate the requested capacity of the snapshot at the beginning only once.
 	if vmSnapshot.Status.Storage.Requested == nil {
-		ctx.Logger.V(4).Info("Updating snapshot's status requested capacity")
-		vmSnapshot.Status.Storage.Requested = []vmopv1.VirtualMachineSnapshotStorageStatusRequested{}
-		requested, err := kubeutil.CalculateReservedForSnapshotPerStorageClass(ctx, r.Client, r.Logger, *vmSnapshot)
-		if err != nil {
-			return ctrl.Result{}, fmt.Errorf("failed to calculate requested capacity for snapshot: %w", err)
-		}
-		vmSnapshot.Status.Storage.Requested = requested
-		ctx.Logger.V(5).Info("Updated vmSnapshot requested capacity",
-			"requested", vmSnapshot.Status.Storage.Requested)
-		// Enqueue the storage classes to sync corresponding SPU.
-		for _, requested := range vmSnapshot.Status.Storage.Requested {
-			ctx.StorageClassesToSync.Insert(requested.StorageClass)
+		if err := r.calculateRequestedCapacity(ctx); err != nil {
+			return ctrl.Result{}, err
 		}
 	}
 
-	ctx.Logger.V(5).Info("Updating snapshot's status used capacity")
-	size, err := r.VMProvider.GetSnapshotSize(ctx, vmSnapshot.Name, vm)
-	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("failed to get snapshot size: %w", err)
+	// Only start calculating the used capacity if the snapshot is created.
+	if !pkgcnd.IsTrue(vmSnapshot, vmopv1.VirtualMachineSnapshotCreatedCondition) {
+		return ctrl.Result{}, nil
 	}
-	// TODO(lubron): Now the format will very likely to fall back to decimalSI
-	// since the size is int64 and it's not a multiple of 1024.
-	// We can think of refactoring to show a more humain readble value without loosing
-	// the precision in the future.
-	total := kubeutil.BytesToResource(size)
 
-	if vmSnapshot.Status.Storage.Used == nil {
-		vmSnapshot.Status.Storage.Used = resource.NewQuantity(0, resource.BinarySI)
+	if err := r.calculateUsedCapacity(ctx); err != nil {
```

Comment:
Can you remind me when we are decrementing the requested capacity?  We should do that when the CSI sync is done (creation is not sufficient).

Reply:
In SPU controller, it will not include Snapshot as part of requested capacity if it has CSIVolumeSync annotation set to `Completed`.

On VMSnapshot, we record both reserved and usage, so it would be easier for SPU controller to calculate the total number

Reply:
Got it.  So, we clear out (decrement) the reserved capacity in in the SPU if the snapshot is completed.  Thanks!